### PR TITLE
Remove references to "process" from our runtime types.

### PR DESCRIPTION
**Pull request by spetrovic77**
The term process is a leftover from when we used to have colocation groups hosting multiple processes.

Other changes:
* Get rid of the colocation-group internal transport.
* Small drive-by changes.

### CODE REVIEW

File: .github/workflows/github-pages.yml [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# Copyright 2023 Google LLC
```
Comment (mwhittaker) [inline .github/workflows/github-pages.yml:1]: Oops, thanks for fixing this.

File: runtime/protos/runtime.proto [hunk 2]
```diff
@@ -197,32 +198,30 @@ message ColocationGroup {
...
-  bool single_process = 9;                   // is a single process?
-  bool use_localhost = 10;                   // listeners listen on localhost?
-  bool process_picks_ports = 11;             // a weavelet picks its own port?
-  string network_storage_dir = 12;           // colocation group storage dir
+  string id = 5;                             // weavelet id
+  // TODO(spetrovic): Rename to same_group.
+  repeated ComponentGroup same_process = 6;  // See AppConfig.SameProcess.
+  map<string, string> sections = 7;          // See AppConfig.Sections.
+  bool single_process = 8;                   // a single-process deployment?
+  bool use_localhost = 9;                    // listeners listen on localhost?
```
Comment (mwhittaker) [inline runtime/protos/runtime.proto:210]: Oops, I forgot to delete `use_localhost` and `weavelet_pick_ports` here. I'll fix that in another CL. It's not trivial cause there are still some lingering references in the code.

File: runtime/protos/runtime.proto [hunk 3]
```diff
@@ -145,45 +142,49 @@ message AppConfig {
   // the key is used.
   repeated string env = 6;
 
-  // Groups of components that must be co-located in the same process.
+  // Groups of components that must be co-located together on the same machine
```
Comment (mwhittaker) [inline runtime/protos/runtime.proto:145]: Groups of components that must be co-located together in the same OS process?

File: runtime/protos/runtime.proto [hunk 4]
```diff
@@ -145,45 +142,49 @@ message AppConfig {
...
+  // however, this configuration option guarantees that specified components
+  // will end up in the same group.
   //
   // Components are identified using their full package paths, e.g.:
   //   "github.com/my/project/package/ComponentName"
   //
   // , with the exception of the component associated with the application
   // main(), which is identified using the name "main".
+  //
+  // TODO(spetrovic): Rename to same_group.
```
Comment (mwhittaker) [inline runtime/protos/runtime.proto:176]: Or maybe `colocate` or `colocated`?

File: internal/babysitter/babysitter.go [hunk 5]
```diff
@@ -87,11 +89,11 @@ type Babysitter struct {
 	// statsProcessor tracks and computes stats to be rendered on the /statusz page.
 	statsProcessor *imetrics.StatsProcessor
 
-	mu          sync.RWMutex
-	managed     map[string][]*envelope.Envelope
-	appInfo     *versioned.Map[*AppVersionState]
-	routingInfo *versioned.Map[*protos.RoutingInfo]
-	proxies     map[string]*proxyInfo // proxies, by listener name
+	mu           sync.RWMutex
+	managed      map[string][]*envelope.Envelope
```
Comment (mwhittaker) [inline internal/babysitter/babysitter.go:93]: Mind adding a comment like `// replica envelopes, by group`?

File: internal/babysitter/babysitter.go [hunk 6]
```diff
@@ -24,14 +24,14 @@ import (
 	"net"
```
Comment (mwhittaker) [inline internal/babysitter/babysitter.go:24]: Nice cleanups!

**Review by mwhittaker — APPROVED** [review]
Very thorough! And amazing clean ups too! Thanks Srdjan 🔥🔥 🔥

File: runtime/protos/runtime.proto (hunk 2 above)
Comment (spetrovic77) [reply runtime/protos/runtime.proto:210]: Ack!

File: .github/workflows/github-pages.yml (hunk 1 above)
Comment (spetrovic77) [reply .github/workflows/github-pages.yml:1]: It was auto-fixed for me by build_and_test :)

**Pull request merged**